### ISSUE TITLE
counsel-rg is blocking on Emacs 26 (Release branch)

### ISSUE THREAD

**Issue by gilbertw1** [hlissner/doom-emacs#216]
Hey,

I've noticed that when running counsel-rg/ag commands that Emacs is blocked while the underlying command is actively running, and keystrokes are only registered when the async filter is evaluated.

If I run counsel-rg on a large directory and then type 't' and then press 'e' immediately afterwards, nothing happens for a few moments and the when the results are updated (with additional results from the 'rg' command) the 'e' keypress is registered. Again if I were to press a followup keybinding, it would not register immediately, only when the results in the ivy minibuffer are updated. I found that lowering the `counsel-async-filter-update-time` value makes the input more responsive, which is what lead me to the conclusion that the keystrokes are evaluated when the async filter is run.

I've tracked this down to being caused by:

    (advice-add #'counsel-ag-function :override #'+ivy*counsel-ag-function))

If I comment out ths `advice-add` then no blocking occurs at all. I'll keep looking into what the precise cause of this behavior is, but I wanted to create an issue in case something instantly jumped out at you.

Thanks!

**Comment by hlissner** [comment]
I _think_ I see what you've described when I run `counsel-ag` in, say, `$HOME`, but removing the advice doesn't seem to change anything. Think you could record a gif of it?

That said, `+ivy*counsel-ag-function` only lowers the minimum character limit for searches (from 3 to 1) and auto-escapes the parentheses in the initial query. Besides that, [It](https://github.com/hlissner/doom-emacs/blob/develop/modules/completion/ivy/autoload/ivy.el#L133-L164) and [the function it's replacing](https://github.com/abo-abo/swiper/blob/master/counsel.el#L2009-L2035) are virtually identical.

**Comment by gilbertw1** [comment]
Ah, you are correct. I totally missed the fact that it's lowering the character limit from 3 characters to 1, which is it why felt so much snappier when I commented out the `advice-add` and then I jumped to the conclusion it was your function was the culprit a little too quickly.

Sorry about that, I have not been on the ball today!

For reference I also created this issue on ivy about it: https://github.com/abo-abo/swiper/issues/1222. I'm going to go back and do a little research on this one before I bother anyone else though :)

**Comment by gilbertw1** [comment]
Hey just to follow up on this.

I tracked the problem down to double-buffering on linux which was added after Emacs 25 was released. Double buffering was causing large lag spikes while the `counsel-rg` results buffer was being populated. The theory is that this is much more noticeable on higher resolution displays (my laptop screen is 3840x2160).

I was able to fix the problem by disabling double-buffering using:

    (add-to-list 'default-frame-alist '(inhibit-double-buffering . t))

Also, here's a link to the bug report: https://debbugs.gnu.org/cgi/bugreport.cgi?bug=28695

I think I'd just leave a note here with my findings in case any other linux user experiences the same issue.
